src/contact.rs

```diff
@@ -1026,17 +1026,16 @@ fn cat_fingerprint(
 }
 
 pub fn addr_cmp(addr1: impl AsRef<str>, addr2: impl AsRef<str>) -> bool {
-    let norm1 = addr_normalize(addr1.as_ref());
-    let norm2 = addr_normalize(addr2.as_ref());
+    let norm1 = addr_normalize(addr1.as_ref()).to_lowercase();
```

Comment:
there is a more efficient way to do this, which does not require lowercasing:  https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case

Reply:
the doc says that ` eq_ignore_ascii_case()` is the same as `to_ascii_lowercase(a) == to_ascii_lowercase(b)`. 

not sure if we shouldn't better use `to_lowercase()` instead of `to_ascii_lowercase()` - at least the local part of an email-address may contain weird characters afaik.

Reply:
fair enough, I guess this is an improvement from C, as `strcasecmp` is not fully UTF-8 compatible